modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/QueryPredicateExecutorImpl.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    /** {@inheritDoc} */
+    @Override public Iterable<T> findAll(Predicate pred, OrderSpecifier<?>... orders) {
+        Sort sort = Sort.by(mapQryDslOrderSpecifiersToOrders(orders));
+
+        return executeQuery(prepareSelectQuery(pred), sort);
+    }
+
+    /** {@inheritDoc} */
+    @Override public Iterable<T> findAll(OrderSpecifier<?>... orders) {
+        Sort sort = Sort.by(mapQryDslOrderSpecifiersToOrders(orders));
+
+        return executeQuery(prepareSqlQuery("SELECT * FROM " + type.getSimpleName()), sort);
+    }
+
+    /** {@inheritDoc} */
+    @Override public Page<T> findAll(Predicate pred, Pageable pageable) {
+        return new PageImpl<>(executeQuery(prepareSelectQuery(pred), pageable), pageable, 0);
+    }
+
+    /** {@inheritDoc} */
+    @Override public long count(Predicate pred) {
+        SQLSerializer ser = prepareSqlQuery("SELECT COUNT(*) FROM " + type.getSimpleName() + " WHERE ", pred);
+
+        SqlFieldsQuery qry = new SqlFieldsQuery(ser.toString());
+        qry.setArgs(ser.getConstants().toArray());
+
+        List<List<?>> res = cache.query(qry).getAll();
+
+        return (Long) res.get(0).get(0);
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean exists(Predicate pred) {
+        return !executeQuery(prepareSelectQuery(pred), PageRequest.of(0, 1)).isEmpty();
+    }
+
+    /**
+     * @param pred Predicate.
+     * @return Prepared sql serializer.
+     */
+    private SQLSerializer prepareSelectQuery(Predicate pred) {
+        return prepareSqlQuery(
+            "SELECT * FROM " + type.getSimpleName() + " WHERE ",
+            pred
+        );
+    }
+
+    /**
+     * @param selectPart Select part.
+     * @return Prepared sql serializer.
+     */
+    private SQLSerializer prepareSqlQuery(String selectPart) {
+        return prepareSqlQuery(selectPart, null);
+    }
+
+    /**
+     * @param selectPart Select partition.
+     * @param pred Predicate.
+     * @return Prepared sql serializer.
+     */
+    private SQLSerializer prepareSqlQuery(String selectPart, Predicate pred) {
+        SQLSerializer res = new SQLSerializer(conf).append(selectPart + " ");
+
+        if (pred != null)
+            return res.handle(pred);
+
+        return res;
+    }
+
+    /**
+     * @param ser Serializable.
+     * @param additionalParams Additional params.
+     * @return List of results.
+     */
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private List<T> executeQuery(SQLSerializer ser, Object... additionalParams) {
+        List<Object> params = ser.getConstants();
+        params.addAll(Arrays.asList(additionalParams));
+
+        Object[] arrParams = params.toArray();
+
+        IgniteQuery qry = IgniteQueryGenerator.prepareQuery(type, ser.toString(), arrParams);
+
+        QueryCursor qryCursor = cache.query(QueryUtils.prepareQuery(qry, arrParams));
+
+        return (List<T>) QueryUtils.transformQueryCursor(qryCursor, arrParams, false, ReturnStrategy.LIST_OF_VALUES);
```

Comment:
I think you qryCursor should be wrapped with try-with-resource in order to close cursor after query execution.

Reply:
Done.